client/client.go

```diff
@@ -46,21 +59,23 @@ type Currency string
 // CurrencyUSD is the type of the currency of USC
 const CurrencyUSD = "USD"
 
-// OfferPayload is the term Payload of the price
-type OfferPayload struct {
+// ChargePayload is the charge payload of the offer
+type ChargePayload struct {
 	LeaseContractLength string `json:"leaseContractLength"`
 	PurchaseOption      string `json:"purchaseOption"`
 }
 
 // Offer is the api message of an Offer
 type Offer struct {
-	ID         string
-	InstanceID string
+	ID string
+
+	OfferType  OfferType
+	InstanceID string // If OfferType is not Instance, InstanceID would be nil
```

Comment:
What does the InstanceID look like if OfferType is Instance

Also since it's a string, then the empty value is ""

Reply:
InstanceID may be the skuID of that instance if the type is instance.